admin-part/src/main/kotlin/jvm/Analyzer.kt

```diff
@@ -0,0 +1,207 @@
+package com.epam.drill.plugins.test2code.jvm
+
+import com.epam.drill.plugins.test2code.api.*
+import com.epam.drill.plugins.test2code.common.api.*
+import com.epam.drill.plugins.test2code.coverage.*
+import com.epam.kodux.util.*
+import java.util.*
+
+
+class PackageCoverage(val packageName: String) {
+    var totalInstruction = 0
+    var totalClasses = 0
+    var totalMethods = 0
+    var classes: List<ClassCoverage> = emptyList()
+}
+
+class ClassCoverage(val jvmClassName: String) {
+    var totalInstruction = 0
+    val methods = mutableMapOf<String, MethodCoverage>()
+    var probRangeToInstruction = mutableMapOf<Int, Int>()
+
+    val packageName: String = jvmClassName.substringBeforeLast("/").weakIntern()
+    val className: String = jvmClassName.substringAfterLast("/").weakIntern()
+
+    fun method(name: String, desc: String): MethodCoverage {
+        val methodName = name.weakIntern()
+        return methods[methodName] ?: MethodCoverage(
+            methodName,
+            desc.weakIntern()
+        ).also { methods += (methodName to it) }
+    }
+
+    fun toCoverageUnit(probes: Probes): ClassCounter {
+        val coverageUnits = methods.values.map {
+            it.toCoverageUnit(jvmClassName, probes)
+        }
+        return ClassCounter(
+            path = packageName,
+            name = className,
+            methods = coverageUnits,
+            count = Count(coverageUnits.sumOf { it.count.covered }, totalInstruction)
+        )
+
+    }
+}
+
+class MethodCoverage(val name: String, val desc: String) {
+    var probRangeToInstruction = mutableMapOf<Int, Int>()
+    var totalInstruction: Int = 0
+    fun toCoverageUnit(ownerClass: String, probes: Probes): MethodCounter {
+        var covered = 0
+        probRangeToInstruction.forEach { (probeId, v) -> if (probes[probeId]) covered += v }
+        return weakPool(
+            MethodCounter(
+                name = name,
+                desc = desc,
+                decl = desc,
+//            key = "$ownerClass:$name$desc".weakIntern(), todo what is the key?
```

Comment:
 this one for future.. for covermethod calculation algo.
 will use for hashMap key